### PR TITLE
Clean a bit PaymentModule class

### PR DESCRIPTION
**Pull request by matks**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Clean a bit PaymentModule class. I fix bad english, fix some phpDoc and remove a deprecated constant. And add a return early
| Type?         | refacto
| Category?     | CO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | 
| How to test?  | No need for QA

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/20383)

### CODE REVIEW

**Review by Progi1984 — DISMISSED** [review]

File: classes/PaymentModule.php [hunk 1]
```diff
@@ -817,9 +819,8 @@ public function getCurrency($current_id_currency = null)
         if (!isset($id_currency) || empty($id_currency)) {
             return false;
         }
-        $currency = new Currency((int) $id_currency);
 
-        return $currency;
+        return Currency((int) $id_currency);
```
Comment (Matt75) [inline classes/PaymentModule.php]:
```suggestion
        return Currency::getCurrencyInstance((int) $id_currency);
```

**Comment by Matt75** [comment]
@matks Can you remove all `die(Tools::displayError());` in this class ?
Because `Tools::displayError()` throw a PrestaShopException only when `_PS_MODE_DEV_` is to true so in production it die...
Currently PaymentModule developer cannot catch it to fallback on degraded or error response properly... So it can be a great improvement 👍

**Comment by matks** [comment]
> @matks Can you remove all `die(Tools::displayError());` in this class ?
> Because `Tools::displayError()` throw a PrestaShopException only when `_PS_MODE_DEV_` is to true so in production it die...
> Currently PaymentModule developer cannot catch it to fallback on degraded or error response properly... So it can be a great improvement 👍

I can, but then in another PR.
Because the current PR brings no changes to the behavior (so it can be tested without QA). Removing `die(Tools::displayError());` cannot be done without thorough testing from both me and the QA team

**Comment by Matt75** [comment]
Ok, i can open an issue for that if needed

**Review by PierreRambaud — APPROVED** [review]

**Review by Progi1984 — APPROVED** [review]

**Comment by Progi1984** [comment]
Thanks @matks . No need QA

**Pull request merged**